webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/ScheduledJobManager.java

```diff
@@ -0,0 +1,249 @@
+      }
+
+      scheduledJob.setId(syncJobConfig.getUuid());
+      scheduledJob.setCron(syncJobConfig.getCron());
+      scheduledJob.setRepository(repositoryRepository.findByName(syncJobConfig.getRepository()));
+      scheduledJob.setJobStatus(
+          scheduledJobStatusRepository.findByEnum(ScheduledJobStatus.SCHEDULED));
+      scheduledJob.setJobType(
+          scheduledJobTypeRepository.findByEnum(ScheduledJobType.THIRD_PARTY_SYNC));
+      scheduledJob.setProperties(getScheduledThirdPartySyncProperties(syncJobConfig));
+
+      try {
+        scheduledJobRepository.save(scheduledJob);
+      } catch (DataIntegrityViolationException e) {
+        // Attempted to insert another scheduled job into the table with the same repo and job type,
+        // this can happen in a clustered quartz environment, don't need to display the error.
+      }
+    }
+  }
+
+  /**
+   * Remove jobs defined under this custom scheduler that are not listed in the application
+   * properties but are present in the DB table.
+   */
+  public void cleanQuartzJobs() throws SchedulerException {
+    // Clean Quartz jobs that don't exist in the UUID pool
+    logger.info("Performing Quartz scheduled jobs clean up");
+    for (JobKey jobKey : getScheduler().getJobKeys(GroupMatcher.anyGroup())) {
+      if (!uuidPool.contains(jobKey.getName())) {
+        if (getScheduler().checkExists(jobKey)) {
+          getScheduler().deleteJob(jobKey);
+        }
+
+        scheduledJobRepository
+            .findById(jobKey.getName())
+            .ifPresent(
+                job -> {
+                  scheduledJobRepository.delete(job);
+                });
+
+        logger.info(
+            "Removed job with id: '{}' as it is no longer in the data source.", jobKey.getName());
+      }
+    }
+  }
+
+  // v1
+  private ScheduledThirdPartySyncProperties getScheduledThirdPartySyncProperties(
+      ThirdPartySyncJobConfig jobConfig) {
+    ScheduledThirdPartySyncProperties thirdPartySyncProperties =
+        new ScheduledThirdPartySyncProperties();
+    thirdPartySyncProperties.setThirdPartyProjectId(jobConfig.getThirdPartyProjectId());
+    thirdPartySyncProperties.setActions(jobConfig.getActions());
+    thirdPartySyncProperties.setPluralSeparator(jobConfig.getPluralSeparator());
+    thirdPartySyncProperties.setLocaleMapping(jobConfig.getLocaleMapping());
+    thirdPartySyncProperties.setSkipTextUnitsWithPattern(jobConfig.getSkipTextUnitsWithPattern());
+    thirdPartySyncProperties.setSkipAssetsWithPathPattern(jobConfig.getSkipAssetsWithPathPattern());
+    thirdPartySyncProperties.setIncludeTextUnitsWithPattern(
+        jobConfig.getIncludeTextUnitsWithPattern());
+    thirdPartySyncProperties.setOptions(jobConfig.getOptions());
+    return thirdPartySyncProperties;
+  }
+
+  public Trigger buildTrigger(JobKey jobKey, String cronExpression, TriggerKey triggerKey) {
+    // Misfires should only occur when the job was triggerred manually and missed its schedule,
```

Comment:
If you were to manually trigger the job and the job is executing over its cron schedule e.g the job is executing when the cron trigger tries to fire, when the manually triggered job finishes Quartz will think the original cron schedule misfired and will instantly fire the job off. This really isn't ideal to instantly fire the job again after completion.

Reply:
> This really isn't ideal to instantly fire the job again after completion.

I think this warrants a further discussion as to the user expectations around this, i.e.: if a user changes a translation in the third party TMS and wants those translations synced right away (and they had a button to trigger this job), they wouldn't care that another job is already running for some time (that potentially missed their changes) and will soon finish, they want to know their changes will be brought in as soon as possible - hence an instant fire after completion.
(Currently, the API/WS response would tell them they can't trigger because a job is already running, right?)

CC: @maallen - any thoughts on the desired behaviour for jobs that were triggered while another job is still running? I'm leaning towards an instant fire afterwards (more computation) - to ensure the manual action is respected/actioned vs. giving non-actionable info back to the user/needing them to check later. (This would impact the trigger endpoint behaviour as well - to make sure that is in sync)

Reply:
@garionpin I have changed it so the manual trigger can queue up for a max of one retrigger, meaning the job will retrigger if it was triggered during execution and it will only trigger once even if it was triggered multiple times when it was running.

Reply:
@mattwilshire - thank you!